go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -355,6 +372,70 @@ func (se *Engine) tableWasCreatedOrAltered(ctx context.Context, tableName string
 	return nil
 }
 
+// registerTopics optionally connects the vt_topic metadata on a message table
+// to a map of topic strings. A table can belong to only one topic.
+func (se *Engine) registerTopics() {
+	for _, table := range se.tables {
+		se.registerTopic(table)
+	}
+}
+
+func (se *Engine) registerTopic(ta *Table) {
+	if ta.MessageInfo == nil || ta.MessageInfo.Topic == "" {
+		return
+	}
+
+	topicName := ta.MessageInfo.Topic
+	topicTable, ok := se.tables[topicName]
+	if !ok {
+		// initialize topic table if necessary
+		topicTable = NewTable(topicName)
+		topicTable.TopicInfo = &TopicInfo{
+			Subscribers: make([]*Table, 0, 1),
+		}
+		se.tables[topicName] = topicTable
+		log.Infof("creating topic table '%s'", topicName)
+	} else {
+		// check to see if this table is already registered to the topic
+		// so we don't double register
+		for _, t := range topicTable.TopicInfo.Subscribers {
+			if t.Name == ta.Name {
```

Comment:
is this the best way to handle table equality? I tried pointers at first, but they were different.

Reply:
Table name equality should work. How about rebuilding the topics from scratch every time instead? We can set a flag in the code that reloads if any table has changed, and use that to decide if a rebuild is needed. This will also get rid of `unregisterTopic`.

Reply:
I actually had rebuilding on every reload implemented originally, but switched away from that when I was still trying for pointer equality. I can switch back to that implementation.